### DISCUSSION TITLE
Determine which sensor provided a detection

### DISCUSSION THREAD

**Discussion by faultfactory** [dstl/Stone-Soup#1054] [General]
I'm using StoneSoup to generate unit test data with ground truth for our tracker implementation and we process sensor inputs individually. 
Is there a way to identify which sensor generated a detection? Is there a setting buried deep in there to embed this into the metadata? I'm using PlatformDetectionSimulator, with MultiTransitionMovingPlatforms. 
I'm building my own sensor classes, so I've got the flexibility to do something there by assigning unique IDs.

**Comment by apiszcz** [comment]
One option is to add a dictionary with your platform IDs to
https://stonesoup.readthedocs.io/en/latest/stonesoup.types.html#stonesoup.types.track.Track.metadata

**Comment by sdhiscocks** [comment]
Not currently. You could add the sensor, whether a reference to sensor itself or an ID, to the metadata of the detection whilst generating it in your sensor class.

It'd be useful to add this at it's own property in general I think. Similar to the special "TrueDetection" class we have for simulation has reference to truth it is an observation of.

**Reply by faultfactory** [reply]
You'd need a corresponding sensor identifier as well, correct? 
I'd offer to PR this but my Python leaves much to be desired. more of a C++ guy.

**Comment by faultfactory** [comment]
I'm going forward with the metadata for the time being now that i figured out how to make my own sensor. Thanks for the input. much appreciated.
